### ISSUE TITLE
False positive no-unused-vars on import in custom.d.ts

### ISSUE THREAD

**Issue by ajmas** [eslint/eslint#13388]
**Tell us about your environment**

Node version: v12.16.2
npm version: v6.13.6
Local ESLint version: v6.8.0 (Currently used)
Global ESLint version: Not found
OS: macOS 10.15.5

**What parser (default, Babel-ESLint, etc.) are you using?**

 default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
    'env': {
        'es6': true
    },
    'extends': 'eslint:recommended',
    'globals': {
        'process': true
    },
    'settings': {
        'import/resolver': {
            'node': {
                'extensions': ['.js', '.jsx', '.ts', '.tsx']
            }
        }
    },
    'parser': '@typescript-eslint/parser',
    'parserOptions': {
        'project': 'tsconfig.json',
        'sourceType': 'module'
    },
    'plugins': [
        '@typescript-eslint',
        'eslint-plugin-import'
    ],
    'rules': {
        '@typescript-eslint/indent': ['error', 4],
        '@typescript-eslint/no-require-imports': 'error',
        '@typescript-eslint/quotes': [
            'error',
            'single'
        ],
        'comma-dangle': 'error',
        'curly': 'error',
        'eqeqeq': [
            'error',
            'always'
        ],
        'no-trailing-spaces': 'error',
        'no-var': 'error',
        'prefer-const': 'error',
        'no-console': 'error',
        'no-unused-vars': ['error', { 'args': 'none' }],
        'no-throw-literal': 'error',
        'no-trailing-spaces': 'error',
        'import/no-extraneous-dependencies': 'error',
        'import/no-cycle': 'error',
        'import/no-self-import': 'error',
        'import/no-unresolved': 'error'
    },
    'overrides': [{
        'files': ['*.ts', '*.tsx'],
        'rules': {
            '@typescript-eslint/no-unused-vars': [2, { args: 'none' }],
            'no-undef': 'off'
        }
    }]
};

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

Contents of custom.d.ts file:
```js
/* eslint-disable no-unused-vars */
import { Server } from 'ws';

declare global {
    namespace Express {
        interface Request {
            ws?: Server;
        }
    }
}

export {};
```

```bash
eslint -c .eslintrc.js --ext .ts src
```

**What did you expect to happen?**

I expected to be able to define the type of ws (imported from @types/ws)

**What actually happened? Please include the actual, raw output from ESLint.**

Two different failures:

  - the import is considered unused
  - `eslint-disable no-unused-vars` has no effect
 
**Are you willing to submit a pull request to fix this bug?**

I don't have a fix to offer at this point.

**Comment by yeonjuan** [comment]
@ajmas 
Hi :)  some eslint rule does not work properly with typescript code.
What about use [typescript-eslint/no-unused-vars](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-unused-vars.md) instead?

```js
    'overrides': [{
        'files': ['*.ts', '*.tsx'],
        'rules': {
            'no-unused-vars': 'off', // off the eslint core rule.
            '@typescript-eslint/no-unused-vars': [2, { args: 'none' }],
            'no-undef': 'off'
        }
    }]

```

**Comment by ajmas** [comment]
Getting an equivalent error with the Typescript rule:

> 'Server' is defined but never used.eslint@typescript-eslint/no-unused-vars

The updated config:

```js
module.exports = {
    'env': {
        'es6': true
    },
    'extends': 'eslint:recommended',
    'globals': {
        'process': true
    },
    'settings': {
        'import/resolver': {
            'node': {
                'extensions': ['.js', '.jsx', '.ts', '.tsx']
            }
        }
    },
    'parser': '@typescript-eslint/parser',
    'parserOptions': {
        'project': 'tsconfig.json',
        'sourceType': 'module'
    },
    'plugins': [
        '@typescript-eslint',
        'eslint-plugin-import'
    ],
    'rules': {
        '@typescript-eslint/no-unused-vars': [2, { args: 'none' }],
        '@typescript-eslint/indent': ['error', 4],
        '@typescript-eslint/no-require-imports': 'error',
        '@typescript-eslint/quotes': [
            'error',
            'single'
        ],
        'comma-dangle': 'error',
        'curly': 'error',
        'eqeqeq': [
            'error',
            'always'
        ],
        'no-trailing-spaces': 'error',
        'no-var': 'error',
        'prefer-const': 'error',
        'no-console': 'error',
        'no-throw-literal': 'error',
        'no-trailing-spaces': 'error',
        'import/no-extraneous-dependencies': 'error',
        'import/no-cycle': 'error',
        'import/no-self-import': 'error',
        'import/no-unresolved': 'error',
        'no-undef': 'off',
        'no-unused-vars': 'off'
    },
    'overrides': [{
        'files': ['*.ts', '*.tsx'],
        'rules': {
            '@typescript-eslint/no-unused-vars': [2, { args: 'none' }],
            'no-undef': 'off'
        }
    }]
};
```

**Comment by ajmas** [comment]
At this point, given there is no resolution, I would like to treat this as a bug.

**Comment by mdjermanovic** [comment]
This seems to be a problem with `@typescript-eslint/no-unused-vars` rule, which is from `@typescript-eslint/eslint-plugin`.

Could you please open an issue in [typescript-eslint/typescript-eslint](https://github.com/typescript-eslint/typescript-eslint) repository?

**Comment by ajmas** [comment]
@mdjermanovic thanks. Will do.
